### ISSUE TITLE
allow user to read builder config and specifically constraints

### ISSUE THREAD

**Issue by michelmno** [openSUSE/open-build-service#1012]
allow user to read builder config and specifically constraints
I understand that today only admins are able to access builder configurations.

my use case as initially documented in (1) is to allow user to understand
why scheduled tasks are not dispatched to builders that seems idles in builder status (2)

(1) http://lists.opensuse.org/opensuse-buildservice/2015-08/msg00004.html
(2) https://build.opensuse.org/monitor

---
Michel Normand

**Comment by hennevogel** [comment]
So the situation is that there are build constraints that constantly can't be satisfied with the available workers. In this situation there is no error reporting only constant rescheduling

**Comment by k0da** [comment]
and worker capabilities are not exposed to public

**Comment by hennevogel** [comment]
Why would they need to be?

**Comment by k0da** [comment]
I could imagine the following usecase:
buildjob is failing. No space left on device, I wonder why, then go and look what was the build host configuration.

**Comment by hennevogel** [comment]
Why? You won't have any influence on which host get's picked the next time anyway. Maybe we could expose this to the admins, but not to users.

**Comment by k0da** [comment]
You can adapt _constraints to get your job scheduled to the right worker, without guessing a result

**Comment by hennevogel** [comment]
_constraints are not meant to get your job scheduled like you want to :neckbeard:

**Comment by adrianschroeter** [comment]
yes, we need a dispatch state/error reporting. We discussed it several times meanwhile, but have no concrete plan for that. Will take care ...

We can also think about how to make work capabilities available, but that is another feature.
